service/src/java/org/apache/hive/service/cli/thrift/ThriftBinaryCLIService.java

```diff
@@ -105,4 +140,79 @@ public void run() {
+     ExecutorService executorService = new ThreadPoolExecutor(minThreads, maxThreads,
+         service.workerKeepAliveTime, TimeUnit.SECONDS, new SynchronousQueue<Runnable>(),
+         new ThreadFactoryWithGarbageCleanup(threadPoolName));
+
+     int customPortNum;
+     String portString = System.getenv("HIVE_SERVER2_KERBEROS_CUSTOM_AUTH_PORT");
+     if (portString != null) {
+       customPortNum = Integer.valueOf(portString);
+     } else {
+       customPortNum = hiveConf.getIntVar(ConfVars.HIVE_SERVER2_KERBEROS_CUSTOM_AUTH_PORT);
+     }
+
+     HiveAuthFactory hiveAuthFactory = new HiveAuthFactory(hiveConf);
+     TTransportFactory transportFactory = hiveAuthFactory.getAuthPlainTransFactory();
+     TProcessorFactory processorFactory = hiveAuthFactory.getAuthProcFactory(service);
+     TServerSocket customKrbSocket = null;
+
+     if (!hiveConf.getBoolVar(ConfVars.HIVE_SERVER2_KERBEROS_CUSTOM_AUTH_SSL_USED)) {
+       customKrbSocket = HiveAuthFactory.getServerSocket(hiveHost, customPortNum);
+     } else {
+       List<String> sslVersionBlacklist = new ArrayList<String>();
+       for (String sslVersion : hiveConf.getVar(ConfVars.HIVE_SSL_PROTOCOL_BLACKLIST).split(",")) {
+         sslVersionBlacklist.add(sslVersion);
+       }
+       String keyStorePath = hiveConf.getVar(ConfVars.HIVE_SERVER2_KERBEROS_CUSTOM_AUTH_SSL_KEYSTORE_PATH).trim();
+       if (keyStorePath.isEmpty()) {
+         throw new IllegalArgumentException(ConfVars.HIVE_SERVER2_KERBEROS_CUSTOM_AUTH_SSL_KEYSTORE_PATH.varname +
+         " Not configured for SSL connection");
+       }
+       String keyStorePassword = ShimLoader.getHadoopShims().getPassword(hiveConf,
+           HiveConf.ConfVars.HIVE_SERVER2_KERBEROS_CUSTOM_AUTH_SSL_KEYSTORE_PASSWORD.varname);
+       customKrbSocket = HiveAuthFactory.getServerSSLSocket(hiveHost, customPortNum,
+         keyStorePath, keyStorePassword, sslVersionBlacklist);
+     }
+
+     // Server args
+     int maxMessageSize = hiveConf.getIntVar(HiveConf.ConfVars.HIVE_SERVER2_THRIFT_MAX_MESSAGE_SIZE);
+     int requestTimeout = (int) hiveConf.getTimeVar(
+         HiveConf.ConfVars.HIVE_SERVER2_THRIFT_LOGIN_TIMEOUT, TimeUnit.SECONDS);
+     int beBackoffSlotLength = (int) hiveConf.getTimeVar(
+         HiveConf.ConfVars.HIVE_SERVER2_THRIFT_LOGIN_BEBACKOFF_SLOT_LENGTH, TimeUnit.MILLISECONDS);
+     TThreadPoolServer.Args sargs = new TThreadPoolServer.Args(customKrbSocket)
+         .processorFactory(processorFactory).transportFactory(transportFactory)
+         .protocolFactory(new TBinaryProtocol.Factory())
+         .inputProtocolFactory(new TBinaryProtocol.Factory(true, true, maxMessageSize, maxMessageSize))
+         .requestTimeout(requestTimeout).requestTimeoutUnit(TimeUnit.SECONDS)
+         .beBackoffSlotLength(beBackoffSlotLength).beBackoffSlotLengthUnit(TimeUnit.MILLISECONDS)
+         .executorService(executorService);
+
+     // TCP Server
+     customKrbServer = new TThreadPoolServer(sargs);
```

Comment:
I am a little nervous about this assignment because you are assigning to a stack variable.
will this object get garbage collected?  that's not what you intend.  you intend to set the member of the ThriftBinaryCLIService instance.  right?    definitely change the function signature here.